Separating RPI, RPI3 and RPI4 documentation

pavlenex opened the pull request:
This PR is not my work. I took hard work of @normandmickey and @wiz and tried to structure it a bit better considering that we now have two separate guides for RPI.

This PR
1. Removes RPI3 instructions from RaspberryPiDeployment.md and adds generic description and links to RPI3.md and RPI4.md which are now separate pages.
2. Creates RPI3.md previously known as RaspberryPiDeployment.md that @normandmickey created (I credited him as the co-author since the GitHub Desktop only allows co-authors, not authors.
3. Creates RPI3.md entirely written by @wiz. I've re-wrote some parts of it to make it more `formal` and also adjusted his tutorials slightly to be for BTCPay Server only.
4. Creates links in Summary.md
5. Adds links in the readme.md

Please, do a thorough review on this one. While I have deployed my BTCPay on RPI3, my experience with RPI is below average level. Also, if there's a better way to credit the original commiters to this PR, please let me know.

dennisreimann commented on RPi3.md:
```diff
@@ -0,0 +1,278 @@
+
+For your Bitcoin experience to be truly self-sovereign and trustless, you should run nodes on your own hardware and internet connection. BTCPayServer is an excellent way to run Bitcoin & Lightning nodes together. Not only are you validating transactions, you also get the ability to accept Bitcoin (layer 1) and Lightning (layer 2) payments.
+
+To that end, here are instructions to install and host your very own BTCPayServer on a Raspberry Pi. We call it a BTCPi.
```
Maybe we can use this introduction sentence to stretch the fact we are talking about the RPI3 here. We should also mention, that the RPI4 is the recommended way to go and lprovide a link to the other docs, similar to the RPI4 page linking to this one.

dennisreimann commented on RPi3.md:
````diff
@@ -0,0 +1,278 @@
+sudo apt install ufw
+```
+
+This command allows SSH connections from your LAN only. Replace `192.168.1.0` with your own subnet:
````
We should highlight that last sentence – I remember messing it up by not paying enough attention 🙈

dennisreimann commented on RPi4.md:
```diff
@@ -0,0 +1,224 @@
+apt install -y ufw
+ufw default deny incoming
+ufw default allow outgoing
+ufw allow 22/tcp
```
This allows SSH connections from anywhere, which should not be the recommended option. Even though it's more error-prone (see my comment in the RPI3 section on this), we should restrict this to the local network. (i.e. `ufw allow from `192.168.1.0/24 to any port 22`)

pavlenex replied:
Ping @wiz thoughts?

wiz replied:
Firewall rules are probably out of scope for BTCPay installation anyway

dennisreimann commented on RPi3.md:
````diff
@@ -0,0 +1,278 @@
+
+Set your environment variables. Make sure the `BTCPAY_HOST` value uses your own domain & subdomain. As usual, run each command separately:
+```bash
+export BTCPAY_HOST="btcpay.YourDomain.com"
````
WIth a [recent PR](#314) we added some more details on the local deployment (see [this part of the diff](https://github.com/btcpayserver/btcpayserver-doc/pull/328/files#diff-76a73d0660e25e9341734f7560c49f19L233)). This is missing here and in the RPI4 docs – just mentioning it in case we want to integrate it in the new versions as well.

pavlenex replied:
ping @normandmickey would you like to tackle this in a different PR?

pavlenex replied:
ping @normandmickey you want to tackle this in a separate PR or?

dennisreimann commented on RPi4.md:
````diff
@@ -0,0 +1,224 @@
+Configure BTCPay by setting some environment variables:
+
+```
+export BTCPAY_HOST="raspberrypi.local"
````
Also see [this comment](https://github.com/btcpayserver/btcpayserver-doc/pull/328/files#diff-0c203a4921804c810330e6cd31690771R223).

pavlenex commented:
Thanks for reviewing @dennisreimann! My markup editor really messed things up.

> For future changes it'd be best to have one line per sentence
I am not sure I see why this one is needed, that's more of a personal preference? Also take a look at how docs look on gitbook rather than GH.  https://docs.google.com/ Since most people will preview them there.

I've:
- cleaned up whitespaces in RPI 4.
- Adjusted the intro and deleted imo redundant text for docs in RPI3
- Formatted bash properly

## Todo
- [ ] Should we cover the SD card, not only HD in RPI4? How about mentioning something for people who don't want to run on .local?
- [ ] @Wiz can you check the unresolved comments that @dennisreimann posted please?

dennisreimann commented:
> I am not sure I see why this one is needed, that's more of a personal preference?

For the output on GitBook it does not make a difference, but I see these writing/editing/reviewing benefits:
- Writing: Long-winded sentences become easier to spot.
- Editing: Sentences are easier to move around, structure becomes very clear.
- Reviewing: Changes are easier to spot as the diffs are done on a per line basis.

I just wanted to promote this idea, because it helped me to write better docs and have them more managable. Oftentimes people liked this approach and [adopted it](zkSNACKs/WasabiDoc#55), nevertheless: Yes, in the end it's a personal preference.

pavlenex commented:
I don't know, I'm personally not fond of that idea and don't see much difference, but perhaps it's just me, if you want to submit a PR and we can discuss it, the way you suggest it all docs would need to be changed, and it's too much pain for too little benefit **IMO**.

britttttk commented:
> For future changes it'd be best to have one line per sentence
> I just wanted to promote this idea

Fair comment, will keep in mind for docs :)

pavlenex commented:
@wiz @NicolasDorier can you jump in and review the changes suggested by @dennisreimann when you have time?

escapedcat commented on RPi4.md:
````diff
@@ -0,0 +1,222 @@
+passwd pi
+```
+
+![](C:\Users\Pavle\Documents\GitHub\btcpayserver-doc\btcpayserver-doc\btcpayserver-doc\img\RPI4Terminal4.png)
````
@pavlenex here a local image path slipped in, which results in a broken image. Would be nice to have it unbroken ;)

dennisreimann replied:
Fixed, thanks for spotting that! 467093ba155042224623687e6bfad9e54c8a3e70
